src/structures.rs

```diff
@@ -180,4 +272,31 @@ mod tests {
         assert_eq!(4, timetable.footpaths[2][0].to);
         assert_eq!(2, timetable.footpaths[4][0].to);
     }
+
+    #[test]
+    fn builder() {
+        let mut b = Timetable::builder();
+        b.trip();
+        assert_eq!(1, b.trips.len());
+        assert_eq!(0, b.stop("a"));
+        assert_eq!(0, b.stop("a"));
+        assert_eq!(1, b.stop("b"));
+    }
+
+    #[test]
+    fn builder_transform() {
+        let mut b = Timetable::builder();
+        b.trip()
+            .s("a", "0:10")
+            .s("b", "0:20")
+            .s("c", "0:30")
+            .trip()
+            .s("b", "0:00")
+            .s("d", "0:40");
```

Comment:
I'd prefers trip() returning a trip builder object instead of self, but that's really subjective.

Reply:
That means an object that will keep a reference to the stations’ map?
I’m really not that comfortable yet with the lifetimes ;)

Reply:
Yep. That's a good reason to improve your lifetime skills!